GraknConsole.java

```diff
@@ -17,145 +17,53 @@
+            System.exit(1);
         }
-    }
+        LineReader reader = LineReaderBuilder.builder().terminal(terminal).build();
+        Printer printer = new Printer(System.out, System.err);
 
-    private void printHelp(PrintStream sout) {
-        HelpFormatter formatter = new HelpFormatter();
-        OutputStreamWriter writer = new OutputStreamWriter(sout, Charset.defaultCharset());
-        PrintWriter printer = new PrintWriter(new BufferedWriter(writer));
-        formatter.printHelp(
-                printer, formatter.getWidth(), "grakn console [options]", null,
-                options, formatter.getLeftPadding(), formatter.getDescPadding(), null
-        );
-        printer.flush();
+        Console console = new Console(options, reader, printer);
+        console.run();
     }
 
-    /**
-     * Invocation from bash script './grakn console'
-     */
-    public static void main(String[] args) {
-        String action = args.length > 1 ? args[1] : "";
-        if(action.equals("version")){
-            System.out.println(Version.VERSION);
-            System.exit(0);
-        }
-
+    private static CommandLineOptions parseCommandLine(String[] args) {
+        CommandLineOptions options = new CommandLineOptions();
+        CommandLine command = new CommandLine(options);
         try {
-            GraknConsole console = new GraknConsole(Arrays.copyOfRange(args, 1, args.length), System.out, System.err);
-            console.run();
-            System.exit(0);
-        } catch (GraknConsoleException e) {
-            System.err.println(e.getMessage());
-            System.err.println("Cause: " + e.getCause().getClass().getName());
-            System.err.println(e.getCause().getMessage());
-            System.exit(1);
-
-        } catch (GraknClientException e) {
-            // TODO: don't do if-checks. Use different catch-clauses by class
-            if (e.getMessage().startsWith(Status.Code.UNAVAILABLE.name())) {
-                System.err.println(ErrorMessage.COULD_NOT_CONNECT.getMessage());
+            command.parseArgs(args);
+            if (command.isUsageHelpRequested()) {
+                command.usage(command.getOut());
+                System.exit(0);
+            } else if (command.isVersionHelpRequested()) {
+                command.printVersionHelp(command.getOut());
+                System.exit(0);
             } else {
-                e.printStackTrace(System.err);
+                return options;
+            }
+        } catch (CommandLine.ParameterException ex) {
+            command.getErr().println(ex.getMessage());
+            if (!CommandLine.UnmatchedArgumentException.printSuggestions(ex, command.getErr())) {
+                ex.getCommandLine().usage(command.getErr());
             }
-            System.exit(1);
-
-        } catch (Exception e) {
-            e.printStackTrace(System.err);
             System.exit(1);
         }
+        return null;
```

Comment:
interesting - is this required? Is java/intelliJ clever enough to realise that we always either return an object or `System.exit(1)`?

Reply:
Yes that's required, unfortunately.